xcodeproj/internal/opts.bzl

```
@@ -659,8 +661,9 @@ def _create_cc_compile_params(
     ]
 
     params = actions.declare_file(
-        "{}.rules_xcodeproj.{}.compile.params".format(
+        "{}.rules_xcodeproj-{}.{}.compile.params".format(
```

I do wonder if the pattern could be switch to increase clarity just a bit. Mostly leaving the static `rules_xcodeproj` value by itself.
```suggestion
        "{}-{}.rules_xcodeproj.{}.compile.params".format(
```

---

I think `YourTargetTame-GeneratorTargetName.rules_xcodeproj.more_stuff` is more confusing than `YourTargetName.rules_xcodeproj-GeneratorTargetName.more_stuff`. This way you can sort of ignore everything from `.rules_xcodeproj` on. How about I change the `-` to a `.` so it's `YourTargetName.rules_xcodeproj.GeneratorTargetName.more_stuff`?

---

I like that too. 👍